src/drivers/uavcan/sensors/flow.cpp

```diff
@@ -76,10 +76,11 @@ void UavcanFlowBridge::flow_sub_cb(const uavcan::ReceivedDataStructure<com::hex:
 
 	flow.quality = msg.quality;
 
-	if (PX4_ISFINITE(msg.rate_gyro_integral[0]) && PX4_ISFINITE(msg.rate_gyro_integral[1])) {
+	if (PX4_ISFINITE(msg.rate_gyro_integral[0]) && PX4_ISFINITE(msg.rate_gyro_integral[1])
+	    && PX4_ISFINITE(msg.rate_gyro_integral[2])) {
 		flow.delta_angle[0] = msg.rate_gyro_integral[0];
 		flow.delta_angle[1] = msg.rate_gyro_integral[1];
-		flow.delta_angle[2] = 0.f;
+		flow.delta_angle[2] = msg.rate_gyro_integral[2];
```

Comment:
Set just this axis to NAN and then we could handle it either in ekf2 or when passing through sensors/vehicle_optical_flow?

```suggestion
		flow.delta_angle[2] = NAN;
```
https://github.com/PX4/PX4-Autopilot/blob/42df7474674b647909b7aed0780b1dd1d41997fc/src/modules/sensors/vehicle_optical_flow/VehicleOpticalFlow.cpp#L129-L162

Reply:
Yes, I'll do that